CHANGES.rst

```diff
@@ -43,6 +43,15 @@ astropy.io.fits
 - Added ``ver`` attribute to set the ``EXTVER`` header keyword to ``ImageHDU``
   and ``TableHDU``. [#6454]
 
+- The performance for reading FITS tables has been significantly improved, in
+  particular when done through ``Table.read``. The ``Table.read`` now also
+  supports a ``memmap=`` keyword argument to indicate whether or not to use
+  memory mapping when reading the table. When reading FITS tables with
+  ``fits.open``, a new keyword argument ``character_as_bytes`` can be passed -
+  when set to `True`, character columns are returned as Numpy byte arrays
+  (Numpy type S) while when set to `False`, the same columns are decoded to
+  Unicode strings (Numpy type U) which uses more memory. [#6821]
```

Comment:
Maybe split in 3 items (performance improvement, memmap, and character_as_bytes ?